autoload/floaterm/wrapper/yazi.vim

```diff
@@ -0,0 +1,50 @@
+" vim:sw=2:
+" ============================================================================
+" FileName: yazi.vim
+" Author: Sam Varga <sam@vargasd.com>
+" GitHub: https://github.com/vargasd
+" ============================================================================
+
+function! floaterm#wrapper#yazi#(cmd, jobopts, config) abort
+  let s:yazi_tmpfile = tempname()
+  let original_dir = getcwd()
+  lcd %:p:h
+
+  let cmdlist = split(a:cmd)
+  let cmd = 'yazi --chooser-file "' . s:yazi_tmpfile . '"'
+  if len(cmdlist) > 1
+    let cmd .= ' ' . join(cmdlist[1:], ' ')
+  else
+    if !has_key(a:config, 'cwd')
+      if glob('%') != ''
+        let cmd .= ' "' . expand('%:p:h') . '"'
```

Comment:
I really like the ranger integration that lands you on the selected file on opening, but I also noticed while implementing this that this is not how most integrations behave, so I'm perfectly open to making this open on the working directory, and customizing my config to open in the buffer directory, especially since yazi doesn't support selecting a file on open.

Reply:
Decided to just go with the standard behavior for consistency.